File: web/client/src/app/shared/model/transaction.ts
export interface TransactionItem {
  id?: number;
  description: string;
  qty: number;
  transaction_id: string;
  article_id: string;
}

export interface Transaction {
  id?: number;
  total: number;
  payment_option: number;
  account_type: number;
  items: TransactionItem[];
  CreatedAt: string;
  transaction_nr: string;
}

File: internal/domain/transaction.go
package domain

import (
	"database/sql"
)

type PaymentOption uint8

const (
	PaymentOptionCash PaymentOption = iota
	PaymentOptionAccount
	PaymentOptionFree
	PaymentOptionSponsor
)

type Transaction struct {
	Base
	Items         []TransactionItem `gorm:"constraint:OnDelete:CASCADE"`
	PaymentOption PaymentOption
	Total         int32
	TotalCredit   int32
	UserID        string         `gorm:"size:36"`
	User          User           `gorm:"foreignKey:UserID"`
	AccountID     sql.NullString `gorm:"size:36"`
	Account       Account        `gorm:"foreignKey:AccountID"`
}

type TransactionItem struct {
	TransactionID string  `gorm:"primaryKey;index;size:36"`
	ProductID     string  `gorm:"primaryKey;index;size:36"`
	Product       Product `gorm:"foreignKey:ProductID"`
	Qty           int8
	Price         int32
}
